cpp/src/cylon/arrow/arrow_comparator.cpp

```diff
@@ -212,15 +212,15 @@ std::shared_ptr<ArrayIndexComparator> CreateArrayIndexComparatorUtil(const std::
 
 template<typename TYPE, bool ASC = true>
 class TwoNumericRowIndexComparator : public TwoArrayIndexComparator {
-  using ARROW_ARRAY_T = typename arrow::TypeTraits<TYPE>::ArrayType;
+  using T = typename TYPE::c_type;
 
  public:
   TwoNumericRowIndexComparator(const std::shared_ptr<arrow::Array> &a1, const std::shared_ptr<arrow::Array> &a2)
-      : arrays({std::static_pointer_cast<ARROW_ARRAY_T>(a1), std::static_pointer_cast<ARROW_ARRAY_T>(a2)}) {}
+      : arrays({a1->data()->template GetValues<T>(1), a2->data()->template GetValues<T>(1)}) {}
 
   int compare(int64_t index1, int64_t index2) const override {
-    auto diff = arrays.at(util::CheckBit(index1))->Value(util::ClearBit(index1))
-        - arrays.at(util::CheckBit(index2))->Value(util::ClearBit(index2));
+    auto diff = arrays.at(util::CheckBit(index1))[util::ClearBit(index1)]
```

Comment:
The compiler might be doing the necessary optimizations already, but can't we use [] instead of at.

Reply:
Not sure whether there is any difference though. Any thoughts?

Reply:
My initial guess is not much of a difference. We can try [] and see.

Reply:
I think this `const T*` is the best! :-)